pkg/controlplane/controller.go

```diff
@@ -257,7 +257,7 @@ func (c *Controller) UpdateKubernetesService(reconcile bool) error {
 		return err
 	}
 	endpointPorts := createEndpointPortSpec(c.PublicServicePort, "https", c.ExtraEndpointPorts)
-	if err := c.EndpointReconciler.ReconcileEndpoints(kubernetesServiceName, c.PublicIP, endpointPorts, reconcile); err != nil {
+	if err := c.EndpointReconciler.ReconcileEndpoints(kubernetesServiceName, c.PublicIP, endpointPorts, true); err != nil {
```

Comment:
do we know why this was this way?
The comment in line 236 explains this is on purpose

> 		// Service definition is not reconciled after first
>		// run, ports and type will be corrected only during
>		// start.

Reply:
anyway, if this has to change, I think that the right place is in Line239 or just removing the (reconcile bool) input variable.

Reply:
In that case, it will also impact master services in line 256:
```
	if err := c.CreateOrUpdateMasterServiceIfNeeded(kubernetesServiceName, c.ServiceIP, servicePorts, serviceType, reconcile); err != nil {
```
but if it's safe, I agree, removing the bool variable is better.

Reply:
:eyes:

Reply:
I think it was related to https://github.com/kubernetes/kubernetes/pull/15791#discussion_r42281891

Reply:
what does it look like to transition the extra ports setting on masters? do we want skewed masters with different settings dueling/churning setting/resetting these, or should the latest master started win?